f5_openstack_agent/lbaasv2/drivers/bigip/resource_manager.py

```diff
@@ -964,6 +992,33 @@ def _attach_profile(self, bigip, vs, profile):
             v.profiles_s.profiles.create(name=profile['name'],
                                          partition=profile['partition'])
 
+    def _attach_http_profile(self, bigip, vs, profile):
+        # If VS is created by legacy Agent instead of Agent Lite, we have to
+        # patch VS profiles property, because /Common/http cannot be dettached.
+        loc = "/" + profile["partition"] + "/" + profile["name"]
+        attached = False
+        payload = {
+            "name": vs['name'],
+            "partition": vs['partition'],
+            "profiles": []
+        }
+        v = self.resource_helper.load(bigip, name=vs['name'],
+                                      partition=vs['partition'])
+        profiles = v.profiles_s.get_collection()
+        for p in profiles:
+            if p.fullPath == loc:
+                attached = True
+                break
+            if p.fullPath == "/Common/http":
+                payload['profiles'].append(loc)
```

Comment:
what if loc is also an HTTP kind profile, of which the profiles will be effective.